src/main/java/at/hannibal2/skyhanni/features/rift/everywhere/PunchcardHighlight.kt

```diff
@@ -0,0 +1,75 @@
+package at.hannibal2.skyhanni.features.rift.everywhere
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.data.HypixelData
+import at.hannibal2.skyhanni.data.IslandType
+import at.hannibal2.skyhanni.events.IslandChangeEvent
+import at.hannibal2.skyhanni.events.LorenzChatEvent
+import at.hannibal2.skyhanni.events.RenderMobColoredEvent
+import at.hannibal2.skyhanni.events.withAlpha
+import at.hannibal2.skyhanni.utils.ColorUtils.toChromaColor
+import at.hannibal2.skyhanni.utils.EntityUtils.isNPC
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.LorenzUtils.isInIsland
+import at.hannibal2.skyhanni.utils.StringUtils.matchMatcher
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import net.minecraft.client.Minecraft
+import net.minecraft.client.entity.EntityPlayerSP
+import net.minecraft.entity.player.EntityPlayer
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+class PunchcardHighlight {
+    private val config get() = SkyHanniMod.feature.rift.punchcard
+    private val ownIGN get() = Minecraft.getMinecraft().thePlayer.name
+    private var lastRiftServer: String = ""
+
+    private val punchedPattern by RepoPattern.pattern(
+        "rift.punchcard.new",
+        "§5§lPUNCHCARD! §r§eYou punched §r§[\\da-z](?:.*?)?(\\w+)§r§[\\da-z] §r§eand both regained §r§a\\+25ф Rift Time§r§e!"
+    )
+
+    val playerList: MutableSet<String> = mutableSetOf()
+
+    @SubscribeEvent
+    fun onRenderMobColored(event: RenderMobColoredEvent) {
```

Comment:
use ```EntityJoinWorldEvent``` instead and ```EntityLivingHelper.setColor```.
I'm currently removing the ```RenderMobColoredEvent``` see #1127 .

Reply:
is there a way to recolor entities with this? so that changing the config would actually apply the new color

Reply:
this also doesn't seem to work correctly with chroma, and for players that joined the server before you

Reply:
The color change is a problem I'm aware. But the recognition of the players should simply work. The issue may be that the tablist is not correctly loaded on the test (event call). So the checks may fail because it things it is not a real Player. Or do I remember it wrong that your logic doesn't use the tablist to detect players?

Reply:
so about the ```EntityJoinWorldEvent```, replace it with ```MobEvent.Spawn.Player```

Reply:
even with `MobEvent.Spawn.Player` the same issue persists with not coloring players that joined the server before you

Reply:
The issue is quite simple. You miss the spawn of those player because the spawn happens before your feature is enabled (Island updates  about 1-2 seconds to late).
To fix that do your logic once at IslandChangeEvent with alle mobs in the Mob Data.Players.